protocol_connect.go

```diff
@@ -1146,15 +1146,20 @@ func (d *connectWireDetail) MarshalJSON() ([]byte, error) {
 		Value string          `json:"value"`
 		Debug json.RawMessage `json:"debug,omitempty"`
 	}{
-		Type:  typeNameFromURL(d.pb.GetTypeUrl()),
-		Value: base64.RawStdEncoding.EncodeToString(d.pb.GetValue()),
+		Type:  typeNameFromURL(d.pbAny.GetTypeUrl()),
+		Value: base64.RawStdEncoding.EncodeToString(d.pbAny.GetValue()),
 	}
 	// Try to produce debug info, but expect failure when we don't have
 	// descriptors.
-	var codec protoJSONCodec
-	debug, err := codec.Marshal(d.pb)
-	if err == nil && len(debug) > 2 { // don't bother sending `{}`
-		wire.Debug = json.RawMessage(debug)
+	if wire.Value != "" { // don't bother sending `{}`
```

Comment:
You'd know better than me, but I think that the zero value of some of the WKTs (like timestamps) produces useful JSON output. I think we'd be okay without this check: sending the zero value of an error detail feels like an uncommon enough case that we don't need to optimize for it.

Reply:
🤷, it's debatable how useful that is. But I had changed it because a value that is two or fewer characters isn't always `{}` with WKTs, too. While it may be a contrived (and not realistic) case, a `google.protobuf.IntValue` could serialize as the JSON value `1`.

I guess I can change the comparison to directly compare the resulting JSON output to `"{}"`, but had done this hoping it was a convenient way to skip the serialization step. (If you see that the `value` property for a WKT is `""`, it is intuitive enough that it's a zero value whose `debug` value is not particularly valuable?)

Reply:
Also, I just realized that a non-zero value for `google.protobuf.Value` could still produce `{}`, so it might be confusing that we just drop the `debug` key in this case. (Admittedly, I can't think of a legit reason to use a error detail of type `google.protobuf.Value`...).

But perhaps the best check is to skip the key if _both_ `wire.Value == "" && serializedJSON == "{}"`.

Reply:
I implemented what I put in that last comment. WDYT?

Reply:
I think my initial urge to skip sending `{}` was misguided. On balance, it seems better to just send the debug info, whatever it may be; I don't want to rely on end users understanding the semantics of zero values and protojson.